tensorflow_datasets/core/download/download_manager.py

```diff
@@ -389,29 +389,7 @@ def manual_dir(self):
     return self._manual_dir
 
 
-# ============================================================================
-# In Python 2.X, threading.Condition.wait() cannot be interrupted by SIGINT,
-# unless it's given a timeout. Here we artificially give a long timeout to
-# allow ctrl+C.
-# This code should be deleted once python2 is no longer supported.
-if sys.version_info[0] > 2:
-
-  def _wait_on_promise(p):
-    return p.get()
```

Comment:
You still need to resolve the promise. The code in `sys.version_info[0] > 2` is executed for Python3 so it should be kept.

Reply:
Okay will make changes. Can you please tell me what is p here I didn't get it.

Reply:
`p` means `promise` from https://pypi.org/project/promise/